cf/weights.py

```diff
@@ -1652,6 +1652,9 @@ def _geometry_ugrid_cells(cls, f, domain_axis, cell_type, auto=False):
         )
 
         for key, aux in auxiliary_coordinates_1d.items():
+            if aux.ctype not in "XYZ":
```

Comment:
The `ctype` can emerge as `None`, e.g. from `type(cf.example_field(1).construct("auxiliarycoordinate2").ctype)`, which leads to two errors emerging in the test suite (`test_weights.py`) of:

```python
...
======================================================================
ERROR: test_weights_polygon_area_geometry (__main__.WeightsTest)
----------------------------------------------------------------------
Traceback (most recent call last):
...
...
...
  File "/home/slb93/git-repos/cf-python/cf/weights.py", line 1655, in _geometry_ugrid_cells
    if aux.ctype not in "XYZ":
TypeError: 'in <string>' requires string as left operand, not NoneType

----------------------------------------------------------------------
```

To avoid the need for type checking, I'd suggest this:

```suggestion
            if str(aux.ctype) not in "XYZ":
```

but any other way to handle the `None` cases is fine.

Reply:
Yes indeed. I'm not sure why I didn't see this, I _thought_ I ran the tests ...